src/metadata_storage_stb_ds.cc

```diff
@@ -493,6 +493,49 @@ bool LocalDestroyBucket(SharedMemoryContext *context, RpcContext *rpc,
   return destroyed;
 }
 
+bool LocalDestroyVBucket(SharedMemoryContext *context, const char *vbucket_name,
+                         VBucketID vbucket_id) {
+  bool destroyed = false;
+  MetadataManager *mdm = GetMetadataManagerFromContext(context);
+  BeginTicketMutex(&mdm->vbucket_mutex);
+  VBucketInfo *info = LocalGetVBucketInfoById(mdm, vbucket_id);
+
+  // TODO(chogan): @optimization Lock granularity can probably be relaxed if
+  // this is slow
+  int ref_count = info->ref_count.load();
+  if (ref_count == 1) {
+    ChunkedIdList *blobs = &info->blobs;
+    BlobID *blobs_arr = (BlobID *)GetIdsPtr(mdm, *blobs);
+    for (u32 i = 0; i < blobs->length; ++i) {
+      blobs_arr[i] = blobs_arr[--blobs->length];
+      break;
+    }
+    ReleaseIdsPtr(mdm);
```

Comment:
I think instead of lines 507-513, we just need to call `FreeIdList(mdm, info->blobs)`, but only if `HasAllocatedBlobs(info)` is `true`.

Reply:
I fixed this in #201.